contractcourt/channel_arbitrator.go

```diff
@@ -1470,8 +1474,19 @@ func (c *ChannelArbitrator) findCommitmentDeadlineAndValue(heightHint uint32,
 		value := htlc.Amt.ToSatoshis()
 		totalValue += value
 
-		if htlc.RefundTimeout < deadlineMinHeight {
-			deadlineMinHeight = htlc.RefundTimeout
+		// Find the expiry height for this outgoing HTLC's incoming
+		// HTLC.
+		deadlineOpt := c.cfg.FindOutgoingHTLCDeadline(htlc)
+
+		// The deadline is default to the current deadlineMinHeight,
+		// and it's overwritten when it's not none.
+		deadline := deadlineMinHeight
+		deadlineOpt.WhenSome(func(d int32) {
+			deadline = uint32(d)
+		})
+
+		if deadline < deadlineMinHeight {
+			deadlineMinHeight = deadline
```

Comment:
Yeah that's what I did initially, and @ProofOfKeags may have a different option here - but I think the idea is to do minimal work in `WhenSome`, ideally we only extract the value and quickly exit this closure.

Reply:
The easiest "formally correct" thing to do here is to use `UnwrapOr`.

```go
deadlineOpt := ...
deadline := uint32(deadlineOpt.UnwrapOr(int32(deadlineMinHeight)))
```

`WhenSome` should be considered a last resort as it is kind of an escape hatch. I included it because it makes "performing an action" with the full option value quite nice, but given that golang is an imperative language, "performing an action" very often means y'all make the choice to mutate some variable which introduces the cognitive overhead of tracking its value *over time*.

Reply:
Catching up on context it looks like this is a highly stateful algorithm where we are gradually updating `deadlineMinHeight`. My original code suggestion still works but it won't really move the needle on the overall ease of understanding the surrounding context, so feel free to ignore it if you'd rather.